reconcile/utils/jobcontroller/controller.py

```diff
@@ -242,19 +248,69 @@ def enqueue_job(
             return True
         return False
 
+    def _lookup_job_uid(self, job_name: str) -> Optional[str]:
+        job_resource = self.oc.get(
+            self.namespace, "Job", job_name, allow_not_found=True
+        )
+        if not job_resource:
+            return None
+        return job_resource.get("metadata", {}).get("uid")
+
+    def build_secret(self, job: K8sJob) -> Optional[V1Secret]:
+        secret_data = job.secret_data()
```

Comment:
Should we label those Job owned Secret with something so they can be easily looked up?